include/ddc/kernels/splines/spline_builder_batched.hpp

```diff
@@ -124,16 +131,57 @@ class SplineBuilderBatched
     template <class Layout>
     void operator()(
             ddc::ChunkSpan<double, spline_domain_type, Layout, memory_space> spline,
-            ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals) const;
+            ddc::ChunkSpan<double, vals_domain_type, Layout, memory_space> vals,
+            std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                    derivs_xmin
+            = std::nullopt,
+            std::optional<ddc::ChunkSpan<double, derivs_domain_type, Layout, memory_space>> const
+                    derivs_xmax
+            = std::nullopt) const;
```

Comment:
What do you think of setting the derivatives as read-only ?
```suggestion
            std::optional<ddc::ChunkSpan<const double, derivs_domain_type, Layout, memory_space>> const
                    derivs_xmin
            = std::nullopt,
            std::optional<ddc::ChunkSpan<const double, derivs_domain_type, Layout, memory_space>> const
                    derivs_xmax
            = std::nullopt) const;
```

Reply:
I think `vals` can also be `ChunkSpan<double const>` right ? If so it should be addressed in another MR because it requires to replace .span_view with .span_cview in several places (I think it should also break gysela).

Reply:
I checked, the original `sll` version is using `const`. There is no automatic sync anymore, I would do the change as soon as possible. In the meantime we do the sync we can do the modifications in Gysela. Regarding the derivatives it should not cause any trouble yet to Gysela.

Reply:
Is 44828b05c3882c325f34106209303fb19a0d03eb ok ?

Reply:
Yes seems good